x-pack/plugins/licensing/server/types.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { BehaviorSubject } from 'rxjs';
+import { TypeOf } from '@kbn/config-schema';
+import { schema } from './schema';
+import { LICENSE_TYPE, LICENSE_STATUS } from './constants';
+import { LicenseFeature } from './license_feature';
+
+/** @public */
+export interface ILicensingCheck {
+  check: LICENSE_STATUS;
+  message?: string;
+}
+/** @public */
+export interface ILicensingPluginSetup {
```

Comment:
I find the name of this a little off from the pattern we've established. The convention we've been using in core is that the name of the interface returned from `setup` is called `*Setup`. That's close to what this service does, but not quite.

I also think we should make the return value of `setup` be more flexible so that we can add additional properties without it being a breaking change.

This is how I envision it:

```ts
interface LicensePluginSetup {
  // `ILicense` would be the name of the interface is currently called `ILicensePluginSetup`
  license$: Observable<ILicense>;
}

class LicensePlugin implements Plugin<LicensePluginSetup> {
  setup() {
    return { license$ };
  }
}
```

Also, can we add TSDocs to each property of this interface?